Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSecurityEventFirewall.yaml

```diff
@@ -0,0 +1,303 @@
+    '%%14597', 'Transport',
+    '%%14598', 'Forward',
+    '%%14599', 'Stream',
+    '%%14600', 'Datagram Data',
+    '%%14601', 'ICMP Error',
+    '%%14602', 'MAC 802.3',
+    '%%14603', 'MAC Native',
+    '%%14604', 'vSwitch',
+    '%%14608', 'Resource Assignment',
+    '%%14609', 'Listen',
+    '%%14610', 'Receive/Accept',
+    '%%14611', 'Connect',
+    '%%14612', 'Flow Established',
+    '%%14614', 'Resource Release',
+    '%%14615', 'Endpoint Closure',
+    '%%14616', 'Connect Redirect',
+    '%%14617', 'Bind Redirect',
+    '%%14624', 'Stream Packet'];
+  let ProtocolTable = datatable (Protocol:int, NetworkProtocol: string)[
+    1, 'ICMP',
+    3, 'GGP',
+    6, 'TCP',
+    8, 'EGP',
+    12, 'PUP',
+    17, 'UDP',
+    20, 'HMP',
+    27, 'RDP',
+    46, 'RSVP',
+    47, 'PPTP data over GRE',
+    50, 'ESP',
+    51, 'AH',
+    66, 'RVD',
+    88, 'IGMP',
+    89, 'OSPF'];
+  let Directions = datatable (DirectionCode:string,NetworkDirection:string, isOutBound:bool)[
+    '%%14592', 'Inbound', false,
+    '%%14593', 'Outbound', true,
+    '%%14594', 'Forward',false,
+    '%%14595', 'Bidirectional', false,
+    '%%14609', 'Listen', false];
+  ///////////////////////////////////////////////////////
+  // this query extract data fields from EventData column from SecurityEvent table
+  ///////////////////////////////////////////////////////
+  let parser = (starttime:datetime=datetime(null), endtime:datetime=datetime(null)
+  , srcipaddr_has_any_prefix:dynamic=dynamic([]), dstipaddr_has_any_prefix:dynamic=dynamic([]), ipaddr_has_any_prefix:dynamic=dynamic([]),dstportnumber:int=int(null)
+  , hostname_has_any:dynamic=dynamic([]), dvcaction:dynamic=dynamic([]),eventresult:string='*', disabled:bool=false
+   ) {
+  let WindowsFirewall_SecurityEvent=
+    (starttime:datetime=datetime(null)
+    , endtime:datetime=datetime(null)
+    , srcipaddr_has_any_prefix:dynamic=dynamic([])
+    , dstipaddr_has_any_prefix:dynamic=dynamic([])
+    , ipaddr_has_any_prefix:dynamic=dynamic([])
+    , dstportnumber:int=int(null)
+    , hostname_has_any:dynamic=dynamic([])
+    , dvcaction:dynamic=dynamic([])
+    , eventresult:string='*'
+    , disabled:bool=false
+      )
+      {    
+      let src_or_any=set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+      let dst_or_any=set_union(dstipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+      let ip_any =set_union(srcipaddr_has_any_prefix, dstipaddr_has_any_prefix, ipaddr_has_any_prefix);
+      let SecurityEventProjected =
+        SecurityEvent
+        | project EventID, EventData, Computer, TimeGenerated, _ResourceId, _SubscriptionId, Type
+      ;
+      // Event IDs between (5151 .. 5159)
+      // will be extracting Event specific fields from 'EventData' field
+      let SecurityEvent_5152 = 
+        SecurityEventProjected 
+        | where (isnull(starttime) or TimeGenerated>=starttime)
+          and  (isnull(endtime) or TimeGenerated<=endtime)
+        | where not(disabled)
```

Comment:
this should be the first filter, even before time filter

Reply:
done